test/AreaEstimator/utest_area_estimator.cpp

```diff
@@ -0,0 +1,151 @@
+#include <gtest/gtest.h>
+
+#include "area_estimator_test_utils.h"
+
+Rectangle cell = read_cell(1, -1, -1, 1);
+
+TEST(AreaEstimator_SmokeTesting, LR_SideAligned_ThroughCenter_Empty) {
+  Ray ray = read_ray(-50, 0, 50, 0);
+  bool test_flag = test_estimator(cell,ray, {0.01, 0.5});
+  EXPECT_TRUE(test_flag);
+}
+
+TEST(AreaEstimator_SmokeTesting, RL_SideAligned_ThroughCenter_Empty) {
+  Ray ray = read_ray(50, 0, -50, 0);
+  bool test_flag = test_estimator(cell,ray, {0.01, 0.5});
+  EXPECT_TRUE(test_flag);
+}
+
+TEST(AreaEstimator_SmokeTesting, DU_SideAligned_ThroughCenter_Empty) {
+  Ray ray = read_ray(0, -50, 0, 50);
+  bool test_flag = test_estimator(cell,ray, {0.01, 0.5});
+  EXPECT_TRUE(test_flag);
+}
+
+TEST(AreaEstimator_SmokeTesting, UD_SideAligned_ThroughCenter_Empty) {
+  Ray ray = read_ray(0, 50, 0, -50);
+  bool test_flag = test_estimator(cell,ray, {0.01, 0.5});
+  EXPECT_TRUE(test_flag);
+}
+
+
+TEST(AreaEstimator_SmokeTesting, DU_CellD_Empty) {
+  Ray ray = read_ray(-2, -2, 2, 2);
+  bool test_flag = test_estimator(cell,ray, {0.01, 0.5});
+  EXPECT_TRUE(test_flag);
+}
+
+TEST(AreaEstimator_SmokeTesting, NearCellCorner_AtAngle_Empty) {
```

Comment:
It's not clear what does "NearCellCorner" and "AtAngle" mean. Try to make these descriptions as precise as possible.